### PR TITLE
DM-44171: Add custom piff logger to suppress non-warnings labeled as warnings

### PR DESCRIPTION
**Pull request by erykoff**

### CODE REVIEW

File: python/lsst/meas/extensions/piff/piffPsfDeterminer.py [hunk 1]
```diff
@@ -272,6 +281,17 @@ class PiffPsfDeterminerTask(BasePsfDeterminerTask):
...
+    def __init__(self, config, schema=None, **kwds):
+        BasePsfDeterminerTask.__init__(self, config, schema=schema, **kwds)
+
+        piffLoggingLevels = {
+            0: logging.CRITICAL,
+            1: logging.WARNING,
+            2: logging.INFO,
+            3: logging.DEBUG,
+        }
+        self.log.setLevel(piffLoggingLevels[self.config.piffLoggingLevel])
```
Comment (jmeyers314) [inline python/lsst/meas/extensions/piff/piffPsfDeterminer.py:293]: This looks like it's setting the level for PiffPsfDeterminer, not piff?  Does this make the log messages [here](https://github.com/lsst/meas_extensions_piff/blob/main/python/lsst/meas/extensions/piff/piffPsfDeterminer.py#L305) and [here](https://github.com/lsst/meas_extensions_piff/blob/main/python/lsst/meas/extensions/piff/piffPsfDeterminer.py#L308) subject to this config setting?

Comment (jmeyers314) [reply python/lsst/meas/extensions/piff/piffPsfDeterminer.py:293]: Can you instead try something like:
```python
piffLogger = logging.getLogger(f'{self.log.name}.piff')
piffLogger.setLevel(...)
```
and then use `piffLogger` when calling `piffResult.fit` ?

File: tests/test_psf.py [hunk 2]
```diff
@@ -264,12 +275,28 @@ def checkPiffDeterminer(self, **kwargs):
...
             exposure=self.exposure
         ).psfCandidates
-        psf, cellSet = self.psfDeterminer.determinePsf(
-            self.exposure,
-            psfCandidateList,
-            metadata,
-            flagKey=self.usePsfFlag
-        )
+
+        logger = logging.getLogger("lsst.psfDeterminer.Piff")
```
Comment (jmeyers314) [inline tests/test_psf.py:279]: Does this name come from `PiffPsfDeterminerTask._DefaultName`?  I don't think I ever knew that before...

**Review by jmeyers314 — CHANGES_REQUESTED** [review]
Looks good.  Just want to make sure we're readjusting levels just for `Piff` and not for `PiffPsfDeterminerTask`.

File: python/lsst/meas/extensions/piff/piffPsfDeterminer.py [hunk 3]
```diff
@@ -120,6 +121,14 @@ class PiffPsfDeterminerConfig(BasePsfDeterminerTask.ConfigClass):
         ),
         default='pixel'
     )
+    piffLoggingLevel = pexConfig.RangeField[int](
+        doc="PIFF-specific logging level, from 0 (least chatty) to 3 (very verbose); "
+            "note that logs will come out with unrelated notations (e.g. WARNING does "
+            "not imply a warning.",
```
Comment (leeskelvin) [inline python/lsst/meas/extensions/piff/piffPsfDeterminer.py:127]: A missing `)` closing parenthesis here.

**Comment by leeskelvin** [comment]
If I read this PR correctly, does this now mean that we won't have any `lsst.finalizeCharacterization.psf_determiner` log messages printed to a standard log output at all by default?

File: python/lsst/meas/extensions/piff/piffPsfDeterminer.py (hunk 1 above)
Comment (erykoff) [reply python/lsst/meas/extensions/piff/piffPsfDeterminer.py:293]: Ah great point!

**Comment by erykoff** [comment]
@leeskelvin I don't know if you're referring to the original PR or now I fixed the problems pointed out by @jmeyers314.
The updated PR will (by default) a) only log from piff what it considers to be the most important messages (called `CRITICAL` but these names don't mean anything in piff-land) b) will log all info and warning messages as usual from the piff psf determiner and other task children, configured with the regular lsst logging configs.
One can also configure to get back the old logging habits of piff.

**Review by jmeyers314 — APPROVED** [review]
Thanks!  All looks good now.

**Pull request merged**